diskqueue.go

```diff
@@ -480,11 +506,17 @@ func (d *diskQueue) persistMetaData() error {
 		d.readFileNum, d.readPos,
 		d.writeFileNum, d.writePos)
 	if err != nil {
-		f.Close()
+		if err := f.Close(); err != nil {
+			return err
```

Comment:
We wanted to return the original error here, but this clobbers that with a different error  from `Close()`.

In some cases, it may be more thorough and robust to handle the error from `Close()`, but in other cases, we do not want to, because there is nothing we can do about it - the only thing we would do is close the file and move on, but the code is already doing that. So while we could add Close error handling in some cases, it will have to be thoughtful, and there may need to be other code changes to make the library continue to work as intended.

Reply:
ok, i will close this pr